zamba/images/config.py

```diff
@@ -403,32 +404,27 @@ def preprocess_labels(cls, values):
         labels = values["labels"]
 
         # lowercase to facilitate subset checking
-        labels["label"] = labels.label.str.lower()
-
-        # one hot encoding
-        labels = pd.get_dummies(labels.rename(columns={"label": "species"}), columns=["species"])
+        labels["class_name"] = labels.label.str.lower()
 
         # We validate that all the images exist prior to this, so once this assembles the set of classes,
         # we should have at least one example of each label and don't need to worry about filtering out classes
         # with missing examples.
-        species_columns = labels.columns[labels.columns.str.contains("species_")]
-        values["species_in_label_order"] = species_columns.to_list()
+        encoder = LabelEncoder()
+        labels["label"] = encoder.fit_transform(labels["class_name"])
+        values["species_in_label_order"] = encoder.classes_.tolist()
 
-        indices = (
-            labels[species_columns].idxmax(axis=1).apply(lambda x: species_columns.get_loc(x))
-        )
-
-        labels["label"] = indices
+        # one hot encoding
+        one_hot = pd.get_dummies(labels, columns=["class_name"])
+        one_hot.columns = one_hot.columns.str.replace("class_name_", "", regex=False)
```

Comment:
Let's use the pd.Series to avoid having to do any replace with `one_hot = pd.get_dummies(labels.class_name)`